aiida/tools/importexport/dbexport/__init__.py

```diff
@@ -138,241 +365,225 @@ def export(
+        msg='`entities` must be specified and given as a list of AiiDA entities',
+    )
     entities = list(entities)
 
-    if type_check(filename, str, allow_none=True) is None:
-        filename = 'export_data.aiida'
-
-    with SandboxFolder() as folder:
-        time_export_start = time.time()
-        export_tree(entities=entities, folder=folder, **kwargs)
-        time_export_end = time.time()
-
-        with tarfile.open(filename, 'w:gz', format=tarfile.PAX_FORMAT, dereference=True) as tar:
-            time_compress_start = time.time()
-            tar.add(folder.abspath, arcname='')
-            time_compress_end = time.time()
+    all_fields_info, unique_identifiers = get_all_fields_info()
 
-    return (time_export_start, time_export_end, time_compress_start, time_compress_end)
+    entities_starting_set, given_node_entry_ids = _get_starting_node_ids(entities, silent)
 
+    (
+        node_ids_to_be_exported,
+        node_pk_2_uuid_mapping,
+        links_uuid,
+        traversal_rules,
+    ) = _collect_export_nodes(given_node_entry_ids, silent, **traversal_rules)
 
-@override_log_formatter('%(message)s')
-def export_tree(
-    entities=None,
-    folder=None,
-    allowed_licenses=None,
-    forbidden_licenses=None,
-    silent=False,
-    include_comments=True,
-    include_logs=True,
-    **kwargs
-):
-    """Export the entries passed in the 'entities' list to a file tree.
+    _check_node_licenses(node_ids_to_be_exported, allowed_licenses, forbidden_licenses)
 
-    .. deprecated:: 1.2.1
-        Support for the parameter `what` will be removed in `v2.0.0`. Please use `entities` instead.
-
-    :param entities: a list of entity instances; they can belong to different models/entities.
-    :type entities: list
+    entries_queries = _get_entry_queries(
+        node_ids_to_be_exported,
+        entities_starting_set,
+        node_pk_2_uuid_mapping,
+        silent,
+        include_comments,
+        include_logs,
+    )
 
-    :param folder: a temporary folder to build the archive before compression.
-    :type folder: :py:class:`~aiida.common.folders.Folder`
+    export_data = _get_export_data(entries_queries, silent)
 
-    :param allowed_licenses: List or function. If a list, then checks whether all licenses of Data nodes are in the
-        list. If a function, then calls function for licenses of Data nodes expecting True if license is allowed, False
-        otherwise.
-    :type allowed_licenses: list
+    # Close progress up until this point in order to print properly
+    close_progress_bar(leave=False)
 
-    :param forbidden_licenses: List or function. If a list, then checks whether all licenses of Data nodes are in the
-        list. If a function, then calls function for licenses of Data nodes expecting True if license is allowed, False
-        otherwise.
-    :type forbidden_licenses: list
+    # note this was originally below the attributes and group_uuid gather
+    check_process_nodes_sealed({
```

Comment:
all of these `check_` functions should probably start with an underscore...

Reply:
yeh but they are in a different file and that would involve depreciations, so going to leave for now